src/inference/dev_api/openvino/runtime/threading/istreams_executor.hpp

```diff
@@ -141,59 +148,50 @@ class OPENVINO_RUNTIME_API IStreamsExecutor : virtual public ITaskExecutor {
+         * @param[in]  name                         The executor name
+         * @param[in]  streams                      @copybrief Config::_streams
+         * @param[in]  threads_per_stream           @copybrief Config::_threads_per_stream
+         * @param[in]  thread_preferred_core_type   @copybrief Config::_thread_preferred_core_type
+         * @param[in]  cpu_reservation              @copybrief Config::_cpu_reservation
+         * @param[in]  streams_info_table           @copybrief Config::_streams_info_table
          */
         Config(std::string name = "StreamsExecutor",
                int streams = 1,
-               int threadsPerStream = 0,
-               ThreadBindingType threadBindingType = ThreadBindingType::NONE,
-               int threadBindingStep = 1,
-               int threadBindingOffset = 0,
-               int threads = 0,
-               PreferredCoreType threadPreferredCoreType = PreferredCoreType::ANY,
-               std::vector<std::vector<int>> streamsInfoTable = {},
-               bool cpuReservation = false)
+               int threads_per_stream = 0,
+               PreferredCoreType thread_preferred_core_type = PreferredCoreType::ANY,
+               bool cpu_reservation = false,
+               std::vector<std::vector<int>> streams_info_table = {})
             : _name{name},
               _streams{streams},
-              _threadsPerStream{threadsPerStream},
-              _threadBindingType{threadBindingType},
-              _threadBindingStep{threadBindingStep},
-              _threadBindingOffset{threadBindingOffset},
-              _threads{threads},
-              _threadPreferredCoreType(threadPreferredCoreType),
-              _streams_info_table{streamsInfoTable},
-              _cpu_reservation{cpuReservation} {}
+              _threads_per_stream{threads_per_stream},
+              _thread_preferred_core_type(thread_preferred_core_type),
+              _cpu_reservation{cpu_reservation},
+              _streams_info_table{streams_info_table} {
+            update_executor_config();
+        }
 
         /**
-         * @brief Modify _streams_info_table and related configuration according to user-specified parameters, bind
-         * threads to cpu cores if cpu_pinning is true.
-         * @param stream_nums Number of streams specified by user
-         * @param threads_per_stream Number of threads per stream specified by user
-         * @param core_type Cpu type (Big/Little/Any) specified by user
-         * @param cpu_pinning Whether to bind the threads to cpu cores
+         * @brief Modify _streams_info_table and related configuration according to configuration
          */
-        void update_executor_config(int stream_nums,
-                                    int threads_per_stream,
-                                    PreferredCoreType core_type,
-                                    bool cpu_pinning);
+        void update_executor_config();
```

Comment:
Do we expect Config users to call this method? If yes please specify in what is the use-cases. If not lets make it private.

Reply:
I think user can call this method. For example, user manually set `streams` of Config to another value which is not initial value, then user should call `update_executor_config()` to update the internal parameters of Config.

Reply:
That sounds like bad user experience. Error prone design since it is easy to forget to call update_executor_config().
If logic assumes some internal data structures need to be updated after one field is changed it means we need to provide correct incapsulation and access rights:
1. All Config fields should become private
2. setter/getter method should use to modify/access private fileds.
3. setter should be responsible for internal structure update (e.g. call set_streams() should call update_executor_config() inside).

Reply:
@dmitry-gorokhov @ilya-lavrenov As we proposed in CVS-130373, the objective of this PR is updating StreamsExecutor configuration and initialization function in both NPU repo and OV repo. Then the rest of changes in other PRs only modify OV repo.